fancy_cache/memory.py

```diff
@@ -55,13 +63,46 @@ def find_urls(urls=None, purge=False):
                 stats = {"hits": hits or 0, "misses": misses or 0}
             yield (url, cache_key, stats)
 
-    if _del_keys:
+    if keys_to_delete:
         # means something was changed
-        for url in _del_keys:
-            remembered_urls.pop(url)
-            misses_cache_key = "%s__misses" % url
-            hits_cache_key = "%s__hits" % url
-            cache.delete(misses_cache_key)
-            cache.delete(hits_cache_key)
 
+        if USE_MEMCACHED_CAS is True:
+            deleted = delete_keys_cas(keys_to_delete)
+            if deleted is True:
+                return
+
+        remembered_urls = cache.get(REMEMBERED_URLS_KEY, {})
+        remembered_urls = delete_keys(keys_to_delete, remembered_urls)
         cache.set(REMEMBERED_URLS_KEY, remembered_urls, LONG_TIME)
+
+
+def delete_keys_cas(keys_to_delete: typing.List[str]) -> bool:
+    result = False
+    tries = 0
+    while result is False and tries < 100:
+        remembered_urls, cas_token = cache._cache.gets(REMEMBERED_URLS_KEY)
+        if remembered_urls is None:
+            return False
+        remembered_urls = delete_keys(keys_to_delete, remembered_urls)
+        result = cache._cache.cas(
+            REMEMBERED_URLS_KEY, remembered_urls, cas_token, LONG_TIME
+        )
+        tries += 1
+    if result is False:
+        LOGGER.error("Fancy cache delete_keys_cas failed after %s tries", tries)
```

Comment:
@peterbe the `LOGGER` variable you mentioned above is used here in the event that the cache key is not saved after 100 tries. 

I could see this as overly defensive code; if you'd prefer to remove these loggers I think that's fine, it's just that it would be nice to know if it does happen in production.

Reply:
Actually, I'm fine with whatever. I almost never use logging because I don't debug production problems by looking at logging. But if you think it's better, let's just go for it. After all you're more of a maintainer than I am at this point :)

Reply:
Thank you! I do use logging a lot on production because it helps me trace back customer issues to specific points in time. For instance, if a customer says "I'm not receiving notifications" I can go back to logs that show when a task was added to Celery and then the result of those tasks. 

I use LogDNA (https://www.logdna.com/) which has been really a great resource for searching/filtering logs.